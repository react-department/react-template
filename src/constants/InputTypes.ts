enum InputTypes {
  Text = 'TEXT',
  Password = 'PASSWORD',
}

export default InputTypes;
