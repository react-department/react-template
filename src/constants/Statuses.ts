enum Statuses {
  idle = 'idle',
  loading = 'loading',
  failed = 'failed',
}

export default Statuses;
