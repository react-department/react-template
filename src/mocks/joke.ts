export default {
  value: 'Chuck Norris can binary search unsorted data.',
};
