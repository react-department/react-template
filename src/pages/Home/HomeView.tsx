import { useTranslation } from 'react-i18next';

import Loader from '../../components/Loader/Loader';
import useFetchJokeQuery from '../../store/slices/jokes/apis/jokesRandom';

import type { ReactElement } from 'react';

import styles from './sass/HomeView.module.scss';
/**
 * Home page view
 */
function HomeView(): ReactElement {
  const { data, refetch: onGetJoke, isFetching } = useFetchJokeQuery();
  const { t } = useTranslation();
  return (
    <div>
      <div className={styles.content}>
        {isFetching ? (
          <Loader />
        ) : (
          <>
            <button type="button" onClick={onGetJoke}>{ t('home.getJoke') }</button>
            <p>{data?.value}</p>
          </>
        )}
      </div>
    </div>
  );
}

export default HomeView;
