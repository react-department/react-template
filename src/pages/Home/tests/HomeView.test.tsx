import { screen, waitFor } from '@testing-library/react';

import { initServer } from '../../../../__tests__/mswServer';
import { render } from '../../../../__tests__/test-utils';
import { testIdLoader } from '../../../constants/TestId';
import joke from '../../../mocks/joke';
import HomeView from '../HomeView';

describe('<HomeView />', () => {
  initServer();
  test('should render component', async () => {
    const { asFragment } = render(<HomeView />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('should render content after loading', async () => {
    render(<HomeView />);
    expect(screen.getByTestId(testIdLoader)).toBeInTheDocument();
    await waitFor(() => expect(screen.queryByTestId(testIdLoader)).not.toBeInTheDocument());
    expect(screen.getByText(joke.value)).toBeInTheDocument();
  });
});
