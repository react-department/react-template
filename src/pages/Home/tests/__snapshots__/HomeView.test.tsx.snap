// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeView /> should render component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="content"
    >
      <div>
        <img
          alt="loader"
          class="appLoader"
          data-testid="loader"
          src="loader.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
